src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,255 @@
+     * Instantiates the Parser object to deal with user input.
+     *
+     * @param command The command input by the user.
+     * @param ui
+     */
+    public Parser(String command, Ui ui) {
+        this.command = command;
+        this.ui = ui;
+    }
+
+    /**
+     * Reads the input given by the user and determines the commands to be carried out.
+     *
+     * @param taskList A collection of tasks.
+     * @return false if command closes the Parser, true otherwise.
+     */
+    public boolean readCommand(TaskList taskList) {
+        Scanner sc = new Scanner(command);
+        String input = sc.nextLine();
+        String[] inputArray = input.split(" ");
+        String first = inputArray[0];
+
+        try {
+            if (input.equals("bye")) {
+                return false;
+            } else if (input.equals("list")) {
+                listTasks(taskList);
+            } else if (first.equals("unmark")) {
+                unmarkTask(input, taskList);
+            } else if (first.equals("mark")) {
+                markTask(input, taskList);
+            } else if (first.equals("todo")) {
+                addTask(input, taskList);
+            } else if (first.equals("deadline")) {
+                addTask(input, taskList);
+            } else if (first.equals("event")) {
+                addTask(input, taskList);
+            } else if (first.equals("delete")) {
+                deleteTask(input, taskList);
+            } else if (first.equals("find")) {
+                findTask(input, taskList);
+            } else {
+                throw new DukeException(LINE + "☹ OOPS!!! I'm sorry, but I don't know what that means :(\n"
+                        + LINE);
+            }
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        } catch (NumberFormatException e) {
+            System.out.println(LINE + "☹ OOPS!!! Please make sure you include the proper index.\n" + LINE);
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println(LINE + "☹ OOPS!!! Please make sure you include the proper index.\n" + LINE);
+        }
+        return true;
+    }
+
+    /**
+     * Returns the lists of tasks onto the console.
+     *
+     * @param taskList A collection of Task objects.
+     */
+    public static void listTasks(TaskList taskList) {
+        int counter = 1;
+        System.out.println(LINE + "Here are the tasks in your list:");
+
+        for (Task t : taskList.getTaskList()) {
+            System.out.println(counter + "." + t.toString());
+            counter++;
+        }
+
+        System.out.println(LINE);
+    }
+
+    /**
+     * Unmarks the task given by the input user command.
+     *
+     * @param input The input command given by the user.
+     * @param taskList A collection of Task objects.
+     * @throws DukeException If input given is invalid.
+     */
+    public static void unmarkTask(String input, TaskList taskList) throws DukeException {
```

Comment:
I really like how you name your methods in this class, very specific as to what it actually does. 👍